### DISCUSSION TITLE
Reduce output size by optimising transforms/css variables

### DISCUSSION THREAD

**Discussion by Adam-Collier** [unocss/unocss#3829] [Q&A]
Hey everyone, I've recently been looking at the output of unoCSS as I felt my CSS file was probably bigger than I should be. And while of course we are only generating classes that have been used in the markup of our site I was wondering if there was a way to take it a step further (if there is already a way to add such functionality mentioned below please let me know).

Essentially I've been looking at the output of the transform classes which feel like the main culprit, a couple of examples below:
```
.translate-y-full {
  --un-translate-y: 100%;
  transform: translate(var(--un-translate-x)) translateY(var(--un-translate-y))
    translateZ(var(--un-translate-z)) rotate(var(--un-rotate))
    rotateX(var(--un-rotate-x)) rotateY(var(--un-rotate-y))
    rotate(var(--un-rotate-z)) skew(var(--un-skew-x)) skewY(var(--un-skew-y))
    scaleX(var(--un-scale-x)) scaleY(var(--un-scale-y))
    scaleZ(var(--un-scale-z));
}
.scale-100 {
  --un-scale-x: 1;
  --un-scale-y: 1;
  transform: translate(var(--un-translate-x)) translateY(var(--un-translate-y))
    translateZ(var(--un-translate-z)) rotate(var(--un-rotate))
    rotateX(var(--un-rotate-x)) rotateY(var(--un-rotate-y))
    rotate(var(--un-rotate-z)) skew(var(--un-skew-x)) skewY(var(--un-skew-y))
    scaleX(var(--un-scale-x)) scaleY(var(--un-scale-y))
    scaleZ(var(--un-scale-z));
}
```

and as you can see there is a lot there which could potentially by optimised. My first thought here is if we could reduce the length of the css variables names by uglifying to two characters or something? My second thought was instead of outputting every transform value each time can we break them out into their individual property?

```
.translate-y-full {
  translate: 100%;
}
.scale-100 {
  scale: 1;
}
```

I'd love to know if this is currently possible and I've just overlooked it or if this is something that could be added :)

**Comment by seahindeniz** [comment]
I would also like to point out that this also solves a rendering bug as well.
Here's a repro https://codepen.io/seahindeniz/pen/yLdjQZm?editors=1000

The top is using the utility classses and the bottom one uses style attribute
![image](https://github.com/user-attachments/assets/c5646485-685e-409b-ba96-563eff4a4d48)
